src/main/java/net/wurstclient/hacks/BuildRandomHack.java

```diff
@@ -218,6 +219,11 @@ private boolean placeBlockLegit(BlockPos pos)
 			if(distanceSqPosVec > eyesPos.squaredDistanceTo(posVec.add(dirVec)))
 				continue;
 			
+			// check if block can be interacted with (chest, crafting table, ...)
+			BlockHitResult hitResult = new BlockHitResult(hitVec, side.getOpposite(), neighbor, false);
+			if (MC.world.getBlockState(neighbor).onUse(MC.world, MC.player, Hand.MAIN_HAND, hitResult).isAccepted())
+				continue;

```

Comment:
This method of detecting if a block can be right-clicked would cause many kinds of blocks to actually perform their right-click actions.